pxfish/field_type.py

```diff
@@ -39,11 +39,26 @@ def add_field_type(*, operation_type, definition, role, path, session):
     else:
         field_type = session.FieldType.new()
         field_type.role = role
-        field_type.array = definition['array']
+        field_type.name = query['name']
         field_type.part = definition['part']
+        field_type.array = definition['array']
         field_type.routing = definition['routing']
-        field_type.name = query['name']
-        field_type.ftype = definition['ftype']
+        # Work around for change in definition file
+        additional_attributes = {
+            'ftype': 'sample',
+            'choices': None,
+            'required': None
+                }
+        for attribute in additional_attributes:
```

Comment:
Can't you just say `additional_attributes.update(definition)`?

Reply:
Though, it would probably be cleaner to merge the definition into a dictionary with the defaults at the top and then use that dictionary throughout instead